src/NServiceBus.Core/Scheduling/Scheduler.cs

```diff
@@ -21,9 +21,12 @@ protected internal override void Setup(FeatureConfigurationContext context)
         {
             context.Settings.Get<Conventions>().AddSystemMessagesConventions(t => typeof(ScheduledTask).IsAssignableFrom(t));
 
-            var defaultScheduler = new DefaultScheduler();
-            context.Container.RegisterSingleton(defaultScheduler);
```

Comment:
I know `DefaultScheduler` is an `internal` class, but do we have any APIs that would let someone get the instance of it from the container anyway? If there is a way, then removing this would be a breaking behavior change.

Reply:
The only api that comes to mind is DisableFeature that has a stringly typed version that would allow internal Features to be manipulated. But I don’t think we have something like this for resolving stuff from the container